crates/burn-core/src/nn/rnn/lstm.rs

```diff
@@ -39,164 +56,253 @@ impl LstmConfig {
     pub fn init<B: Backend>(&self, device: &B::Device) -> Lstm<B> {
         let d_output = self.d_hidden;
 
-        let input_gate = gate_controller::GateController::new(
-            self.d_input,
-            d_output,
-            self.bias,
-            self.initializer.clone(),
-            device,
-        );
-        let forget_gate = gate_controller::GateController::new(
-            self.d_input,
-            d_output,
-            self.bias,
-            self.initializer.clone(),
-            device,
-        );
-        let output_gate = gate_controller::GateController::new(
-            self.d_input,
-            d_output,
-            self.bias,
-            self.initializer.clone(),
-            device,
-        );
-        let cell_gate = gate_controller::GateController::new(
-            self.d_input,
-            d_output,
-            self.bias,
-            self.initializer.clone(),
-            device,
-        );
+        let new_gate = || {
+            GateController::new(
+                self.d_input,
+                d_output,
+                self.bias,
+                self.initializer.clone(),
+                device,
+            )
+        };
 
         Lstm {
-            input_gate,
-            forget_gate,
-            output_gate,
-            cell_gate,
+            input_gate: new_gate(),
+            forget_gate: new_gate(),
+            output_gate: new_gate(),
+            cell_gate: new_gate(),
             d_hidden: self.d_hidden,
         }
     }
 }
 
 impl<B: Backend> Lstm<B> {
     /// Applies the forward pass on the input tensor. This LSTM implementation
-    /// returns the cell state and hidden state for each element in a sequence (i.e., across `seq_length`),
-    /// producing 3-dimensional tensors where the dimensions represent [batch_size, sequence_length, hidden_size].
+    /// returns hidden state for each element in a sequence (i.e., across `seq_length`) and a final state,
+    /// producing 3-dimensional tensors where the dimensions represent `[batch_size, sequence_length, hidden_size]`.
```

Comment:
Since we document the return types, I think we can simplify the docstring here to:

Applies the forward pass on the input tensor. This LSTM implementation returns the state for each element in a sequence (i.e., across `seq_length`) and a final state.